test/metabase/query_processor/middleware/metrics_test.clj

```diff
@@ -716,3 +716,26 @@
                 (mt/format-rows-by
                  [u.date/temporal-str->iso8601-str int int 3.0]
                  (mt/rows (qp/process-query query)))))))))
+
+(deftest ^:parallel expressions-from-metrics-are-spliced-to-correct-stage-test
+  (testing "Integration test: Expression is spliced into correct stage during metric expansion (#48722)"
+    (mt/with-temp [:model/Card source-model {:dataset_query (mt/mbql-query orders {})
+                                             :database_id (mt/id)
+                                             :name "source model"
+                                             :type :model}
+                   :model/Card metric {:dataset_query
+                                       (mt/mbql-query
+                                         orders
+                                         {:source-table (str "card__" (:id source-model))
+                                          :expressions {"somedays" [:datetime-diff
+                                                                    [:field "CREATED_AT" {:base-type :type/DateTime}]
+                                                                    [:now] :day]}
+                                          :aggregation [[:median [:expression "somedays" {:base-type :type/Integer}]]]})
+                                       :database_id (mt/id)
+                                       :name "somedays median"
+                                       :type :metric}]
+      (let [mp (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+            query (-> (lib/query mp (lib.metadata/card mp (:id source-model)))
+                      (lib/aggregate (lib.metadata/metric mp (:id metric))))]
+        (is (= 2162
```

Comment:
Isn't this going to change as time goes by?

Reply:
Thank you, adjusted to fixed datetime.